src/SFA.DAS.AdminService.Web/Controllers/Apply/ApplicationController.cs

```diff
@@ -258,6 +259,142 @@ public async Task<IActionResult> Assessment(Guid applicationId, int sequenceNo,
+                backViewModel.BackAction,
+                backViewModel.BackController,
+                backViewModel.BackOrganisationId,
+                currentVersionIndex);
+
+            return View(nameof(WithdrawalDateCheck), sequenceVm);
+        }
+
+        [HttpPost("/Applications/{applicationId}/{backAction}/{backController}/Sequence/{sequenceNo}/WithdrawalDateCheck/{backOrganisationId?}")]
+        public async Task<IActionResult> WithdrawalDateCheckSave(Guid applicationId, int sequenceNo, BackViewModel backViewModel, string dateApproved, int currentVersionIndex)
+        {
+            var application = await _applyApiClient.GetApplication(applicationId);
+            var organisation = await _apiClient.GetOrganisation(application.OrganisationId);
+
+            var activeApplySequence = application.ApplyData.Sequences.Where(seq => seq.IsActive && !seq.NotRequired).OrderBy(seq => seq.SequenceNo).FirstOrDefault();
+
+            var sequence = await _qnaApiClient.GetSequence(application.ApplicationId, activeApplySequence.SequenceId);
+            var sections = await _qnaApiClient.GetSections(application.ApplicationId, sequence.Id);
+
+            var sequenceVm = new WithdrawalDateCheckViewModel(application, organisation, sequence, sections,
+                activeApplySequence.Sections,
+                backViewModel.BackAction,
+                backViewModel.BackController,
+                backViewModel.BackOrganisationId,
+                currentVersionIndex);
+
+            var errorMessages = new Dictionary<string, string>();
+
+            if (string.IsNullOrWhiteSpace(dateApproved) || (dateApproved.Trim().ToUpper() != "NO" && dateApproved.Trim().ToUpper() != "YES"))
+            {
+                errorMessages["RequestedWithdrawalDate"] = "Please choose an option either 'Yes' or 'No'.";
+            }
+
+            if (errorMessages.Any())
+            {
+                foreach (var error in errorMessages)
+                {
+                    ModelState.AddModelError(error.Key, error.Value);
+                }
+
+                return View(nameof(WithdrawalDateCheck), sequenceVm);
+            }
+
+            if (dateApproved?.Trim().ToUpper() == "NO")
+            {
+                return View(nameof(WithdrawalDateChange), sequenceVm);
+            }
+
+            if (sequenceVm.RequestedWithdrawalDate.HasValue)
+            {
+                if (sequenceVm.SequenceNo == ApplyConst.ORGANISATION_WITHDRAWAL_SEQUENCE_NO)
+                {
+                    await UpdateOrganisationStandardWithdrawalDate(organisation.EndPointAssessorOrganisationId, null, null, sequenceVm.RequestedWithdrawalDate.Value);
+                }
+                else
+                {
+                    await UpdateOrganisationStandardWithdrawalDate(organisation.EndPointAssessorOrganisationId, sequenceVm.StandardReference, sequenceVm.Versions[sequenceVm.CurrentVersionIndex.Value], sequenceVm.RequestedWithdrawalDate.Value);
```

Comment:
Given the bug below found on editing the withdrawal date page, surprised this also didn't error, something to do with the model?